src/System.IO.FileSystem/src/System/IO/CharSpanExtensions.Unix.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace System.IO
+{
+    internal static partial class CharSpanExtensions
+    {
+        internal static unsafe bool EqualsOrdinal(ReadOnlySpan<char> first, ReadOnlySpan<char> second, bool ignoreCase = false)
```

Comment:
@KrzysztofCwalina, @ahsonkhan, another case where we're having to write this logic in multiple places, e.g. here and https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/StringSpanHelpers.cs#L40.  Relates to https://github.com/dotnet/corefx/issues/21395.

Reply:
Yeah, it's totally in the plan to add such APIs. We just need to find some cycles to do it.

Reply:
Thanks.  Just wanted to highlight the proliferation :)